src/UnitTest/test_local_cache.cpp

```diff
@@ -1,215 +1,254 @@
-#else
-    Sleep(expiry_seconds * 1000);
-#endif
-
-    assert(nullptr == local_cache_get(__FUNCTION__));
-
-    TEST_PASSED();
-}
-
-template <typename ExceptionT>
-static void AssertException(void (*function)())
-{
-    try
-    {
-        function();
-    }
-    catch (ExceptionT&)
-    {
-        return;
-    }
-
-    assert(!"Expected exception was not thrown");
-}
-
-//
-// Verify invalid parameter handling
-//
-static void InvalidParams()
-{
-    TEST_START();
-
-    static const uint8_t data[] = "test data";
-
-    AssertException<std::runtime_error>(
-        [] { local_cache_add("", now(), sizeof(data), data); });
-
-    AssertException<std::runtime_error>(
-        []{ local_cache_add(__FUNCTION__, now(), 0, data); });
-
-    AssertException<std::runtime_error>(
-        []{ local_cache_add(__FUNCTION__, now(), sizeof(data), nullptr); });
-
-    AssertException<std::runtime_error>([]{ local_cache_get(""); });
-
-    TEST_PASSED();
-}
-
-//
-// Spawns multiple threads, intentionally creating a lot of contention for
-// a single cache entry.
-//
-static void ThreadSafetyTest()
-{
-    TEST_START();
-
-    // Pre-fill a data vector that's of sufficient size that we
-    // are likely to get conflicts between threads.
-    std::vector<uint8_t> data(64 * 1024);
-    for (size_t i = 0; i < data.size(); ++i)
-    {
-        data[i] = i & 0xff;
-    }
-
-    constexpr unsigned THREAD_LOOP_COUNT = 128;
-    const std::string ID = "data identifier";
-
-    auto cache_writer = [&](void) {
-        for (unsigned i = 0; i < THREAD_LOOP_COUNT; ++i)
-        {
-            local_cache_add(ID, now() + 60, data.size(), data.data());
-        }
-    };
-
-    auto cache_reader = [&](void) {
-        for (unsigned i = 0; i < THREAD_LOOP_COUNT; ++i)
-        {
-            auto retrieved = local_cache_get(ID);
-            assert(retrieved != nullptr);
-            assert(*retrieved == data);
-        }
-    };
-
-    // prime the cache entry first (in case a read thread runs first)
-    local_cache_add(ID, now() + 60, data.size(), data.data());
-
-#if defined(__LINUX__)
-    std::array<std::thread, 8> threads;
-    for (size_t i = 0; i < threads.size(); ++i)
-#else
-    std::thread threads[8];
-    for (size_t i = 0; i < _countof(threads); ++i)
-#endif
-    {
-        if (i & 1)
-        {
-            threads[i] = std::thread(cache_writer);
-        }
-        else
-        {
-            threads[i] = std::thread(cache_reader);
-        }
-    }
-
-    for(auto& t : threads)
-    {
-        t.join();
-    }
-
-    TEST_PASSED();
-}
-
-extern void LocalCacheTests()
-{
-    local_cache_clear();
-
-    AddGetItem();
-    OverwriteCacheEntry();
-    VerifyClearCache();
-    VerifyExpiryWorks();
-    InvalidParams();
-    ThreadSafetyTest();
-}
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+#include <gtest/gtest.h>
+
+#undef NDEBUG // ensure that asserts are never compiled out
```

Comment:
Since we are using gtest now, can this be removed and you replace all the C++ asserts with the test framework one?

Reply:
Yes, that can be removed.